swift/apple/FirezoneNetworkExtension/NetworkSettings.swift

```diff
@@ -9,196 +9,119 @@ import NetworkExtension
-      } else {
-        logger.error("NetworkSettings.apply: Ignoring invalid route '\(route)'")
-      }
-    }
-
+  func apply(completionHandler: (() -> Void)? = nil) {
     // We don't really know the connlib gateway IP address at this point, but just using 127.0.0.1 is okay
     // because the OS doesn't really need this IP address.
     // NEPacketTunnelNetworkSettings taking in tunnelRemoteAddress is probably a bad abstraction caused by
     // NEPacketTunnelNetworkSettings inheriting from NETunnelNetworkSettings.
-
     let tunnelNetworkSettings = NEPacketTunnelNetworkSettings(tunnelRemoteAddress: "127.0.0.1")
 
+    // Set tunnel addresses and routes
     let ipv4Settings = NEIPv4Settings(
-      addresses: [tunnelAddressIPv4], subnetMasks: ["255.255.255.255"])
-    ipv4Settings.includedRoutes = tunnelIPv4Routes
-    tunnelNetworkSettings.ipv4Settings = ipv4Settings
-
-    let ipv6Settings = NEIPv6Settings(addresses: [tunnelAddressIPv6], networkPrefixLengths: [128])
-    ipv6Settings.includedRoutes = tunnelIPv6Routes
-    tunnelNetworkSettings.ipv6Settings = ipv6Settings
-
+      addresses: [tunnelAddressIPv4!], subnetMasks: ["255.255.255.255"])
+    let ipv6Settings = NEIPv6Settings(addresses: [tunnelAddressIPv6!], networkPrefixLengths: [128])
     let dnsSettings = NEDNSSettings(servers: dnsAddresses)
-    // Intercept all DNS queries; SplitDNS will be handled by connlib
+    ipv4Settings.includedRoutes = routes4
+    ipv6Settings.includedRoutes = routes6
     dnsSettings.matchDomains = matchDomains
     dnsSettings.matchDomainsNoSearch = true
+    tunnelNetworkSettings.ipv4Settings = ipv4Settings
+    tunnelNetworkSettings.ipv6Settings = ipv6Settings
     tunnelNetworkSettings.dnsSettings = dnsSettings
     tunnelNetworkSettings.mtu = mtu
 
-    self.hasUnappliedChanges = false
-    logger.log("Attempting to set network settings")
-    packetTunnelProvider.setTunnelNetworkSettings(tunnelNetworkSettings) { error in
+    packetTunnelProvider?.setTunnelNetworkSettings(tunnelNetworkSettings) { error in
       if let error = error {
-        logger.error("NetworkSettings.apply: Error: \(error)")
-      } else {
-        guard !self.hasUnappliedChanges else {
-          // Changes were made while the packetTunnelProvider was setting the network settings
-          logger.log(
-            """
-            NetworkSettings.apply:
-              Applying changes made to network settings while we were applying the network settings
-            """)
-          self.apply(on: packetTunnelProvider, logger: logger, completionHandler: completionHandler)
-          return
-        }
-        logger.log("NetworkSettings.apply: Applied successfully")
+        self.logger.error(
+          "\(#function): Error occurred while applying network settings! Error: \(error.localizedDescription)"
+        )
       }
-      completionHandler?(error)
+
+      completionHandler?()
     }
   }
 }
 
+// For creating IPv4 routes
+enum IPv4SubnetMaskLookup {
```

Comment:
@roop I ended refactoring this to just a lookup table since that's easier to maintain.